iam/snippets/src/main/java/EditRole.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START iam_edit_role]
+
+import com.google.cloud.iam.admin.v1.IAMClient;
+import com.google.iam.admin.v1.Role;
+import com.google.iam.admin.v1.Role.RoleLaunchStage;
+import com.google.iam.admin.v1.UpdateRoleRequest;
+import com.google.protobuf.FieldMask;
+import java.io.IOException;
+
+public class EditRole {
+
+  public static void main(String[] args) {
+    // TODO(developer): Replace the variables before running the sample.
+    // Role ID must point to an existing role.
+    String projectId = "your-project-id";
+    String roleId = "a unique identifier (e.g. testViewer)";
+
+    editRole(projectId, roleId);
+  }
+
+  public static void editRole(String projectId, String roleId) {
+
+    String roleName = "projects/" + projectId + "/roles/" + roleId;
+
+    Role.Builder roleBuilder = Role.newBuilder().setName(roleName).setStage(RoleLaunchStage.GA);
```

Comment:
Instead of editing the launch stage, can we let the user input new information to edit the role? For examples, see other language samples on https://cloud.google.com/iam/docs/creating-custom-roles#edit-role